code/ship/ship.cpp

```diff
@@ -5251,6 +5254,15 @@ static void parse_ship_type(const char *filename, const bool replace)
 		Warning(LOCATION, "Bad ship type name in %s\n\nUsed ship type is redirected to another ship type.\nReplace \"%s\" with \"%s\"\nin %s to fix this.\n", filename, stp->name, ship_type, filename);
 	}
 
+	bool big_ship = false;
+	bool huge_ship = false;
+	if (stricmp(stp->name, "cruiser") == 0 || stricmp(stp->name, "freighter") == 0 || stricmp(stp->name, "transport") == 0 ||
+		stricmp(stp->name, "corvette") == 0 || stricmp(stp->name, "gas miner") == 0 || stricmp(stp->name, "awacs") == 0)
+		big_ship = true;
+
+	if (stricmp(stp->name, "capital") == 0 || stricmp(stp->name, "supercap") == 0 || stricmp(stp->name, "knossos device") == 0 || stricmp(stp->name, "drydock") == 0)
```

Comment:
The `stp->name` field will be "knossos", not "knossos device".  Some of the strings differ between ship classes and ship types, so there is an if() block that standardizes them a few lines up.  You might want to use the `ship_type` string instead of `stp->name`.

Reply:
I'm not sure I follow, thats the entire reason why this was put after the if block, so it could be guaranteed all types that got to this point had standardized names.

Reply:
Oh.  O.o  I guess I had it confused with what happens inside the `ship_type_name_lookup` function.  I see that this is used slightly differently.

That being said, a change is still needed here.  Replace "supercap" with "super cap".

Reply:
Oh man that's super confusing, I got "supercap" from the ship class flags. The hardcoded ship-info flag is "Supercap", the class flag to enable it is "suprecap", but the name of the builtin ship type is "Super cap" and the table will complain if you tried to define "Supercap" because it should be "Super cap".